test/functional/Completion.hs

```diff
@@ -84,6 +84,23 @@ tests = testGroup "completions" [
          compls <- getCompletions doc (Position 5 7)
          liftIO $ assertBool "Expected completions" $ not $ null compls
 
+     , expectFailIfBeforeGhc92 "record dot syntax is introduced in GHC 9.2"
+       $ testGroup "recorddotsyntax"
+        [ testCase "shows field selectors" $ runSession hlsCommand fullCaps "test/testdata/completion" $ do
+            doc <- openDoc "RecordDotSyntax.hs" "haskell"
+
+            let te = TextEdit (Range (Position 25 0) (Position 25 5)) "z = x.a"
+            _ <- applyEdit doc te
+
+            compls <- getCompletions doc (Position 25 6)
+            item <- getCompletionByLabel "a" compls
+            liftIO $ do
+                item ^. label @?= "a"
+                --item ^. detail @?= Just "Data.List"  TODO
```

Comment:
TODO?

Reply:
The TODO was in regards to filling out more of the details field with where the type was defined. It turned out that due to HieAST, the fields' definitions weren't easy to get at (since it would give the types of the fields, but not the fields themselves). I added a comment to Logic.hs about this, and removed the TODO.

Reply:
Maybe worth leaving a todo somewhere since there is something that we might want to improve later?

Reply:
Or even a known broken test!